src/libAtomVM/externalterm.c

```diff
@@ -544,6 +602,20 @@ static int calculate_heap_usage(const uint8_t *external_term_buf, size_t remaini
             return term_boxed_integer_size(value);
         }
 
+        case SMALL_BIG_EXT: {
+            uint8_t num_bytes = external_term_buf[1];
+            if (num_bytes > 8 || remaining < (SMALL_BIG_EXT_BASE_SIZE + num_bytes)) {
+                return INVALID_TERM_SIZE;
+            }
+            uint8_t sign = external_term_buf[2];
+            *eterm_size = SMALL_BIG_EXT_BASE_SIZE + num_bytes;
+            avm_uint64_t value = read_bytes(external_term_buf + 3, num_bytes);
+            if ((sign == 0 && value > INT64_MAX) || (sign == 1 && value > -INT64_MAX)) {
```

Comment:
@bettio see this if clause.

Reply:
sorry, I realize I wasn't really clear, with "testing" I was meaning writing a test case in `tests/erlang_tests/small_big_ext.erl`. And I said checking because I wasn't sure about that.

Reply:
anyway it looks right to me, let's add an UNLIKELY since they are error handling branches.